scripts/generate_multilevel_sensor_constants.py

```diff
@@ -40,46 +50,53 @@ def normalize_name(name: str) -> str:
 
 def format_for_class_name(name: str) -> str:
     """Convert sensor/scale name to class name format."""
-    return normalize_name(name).replace(" ", "")
+    return f"{normalize_name(name).replace(' ', '')}Scale"
 
 
 def normalize_scale_definition(scale_definitions: dict[str, dict]) -> dict[str, int]:
     """Convert a scales definition dictionary into a normalized dictionary."""
-    scale_def = {}
+    scale_def_ = {}
     for scale_id, scale_props in scale_definitions.items():
         scale_id = int(scale_id, 16)
-        scale_name = enum_name_format(scale_props["label"], True)
-        scale_def[scale_name] = scale_id
+        scale_name_ = enum_name_format(scale_props["label"], True)
+        scale_def_[scale_name_] = scale_id
 
-    return scale_def
+    return dict(sorted(scale_def_.items(), key=lambda kv: kv[0]))
 
 
 sensor_types = json.loads(
     remove_comments(
         requests.get(
-            f"https://raw.githubusercontent.com/{GITHUB_PROJECT}/{BRANCH_NAME}/{SENSOR_TYPES_FILE_PATH}"
+            (
+                f"https://raw.githubusercontent.com/{GITHUB_PROJECT}/{BRANCH_NAME}/"
+                f"{SENSOR_TYPES_FILE_PATH}"
+            )
         ).text
     )
 )
 default_scales = json.loads(
     remove_comments(
         requests.get(
-            f"https://raw.githubusercontent.com/{GITHUB_PROJECT}/{BRANCH_NAME}/{DEFAULT_SCALES_FILE_PATH}"
+            (
+                f"https://raw.githubusercontent.com/{GITHUB_PROJECT}/{BRANCH_NAME}/"
+                f"{DEFAULT_SCALES_FILE_PATH}"
+            )
         ).text
     )
 )
 
-scales = {}
-for scale_name, scale_def in default_scales.items():
-    scales[normalize_name(scale_name)] = normalize_scale_definition(scale_def)
+scales = {
+    normalize_name(scale_name): normalize_scale_definition(scale_def)
+    for scale_name, scale_def in default_scales.items()
+}
 
 sensors = {}
 for sensor_id, sensor_props in sensor_types.items():
     sensor_id = int(sensor_id, 16)
     scale_def = sensor_props["scales"]
-    remove_paranthesis_ = True
+    remove_paranthesis_: bool = True
     if sensor_id in (87, 88):
-        remove_paranthesis_ = False
+        remove_paranthesis_: bool = False
```

Comment:
Why is the type needed? Normally those are determined automatically.

Reply:
my editor was complaining but when I ran mypy it didn't so I have removed them